Source/WebCore/bindings/js/JSDOMWindowBase.cpp

```diff
@@ -315,7 +315,11 @@ JSValue toJS(JSGlobalObject* lexicalGlobalObject, DOMWindow& domWindow)
 
 JSDOMWindow* toJSDOMWindow(Frame& frame, DOMWrapperWorld& world)
 {
-    return frame.script().globalObject(world);
+    if (auto* window = JSC::jsDynamicCast<JSDOMWindow*>(frame.script().globalObject(world))) {
+        return window;
+    }
+    ASSERT_NOT_REACHED();
```

Comment:
So this should not be possible, yet we're paying a runtime cost now with the dynamic cast. This doesn't seem ideal.